src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateClaimSpecCallbackHandler.java

```diff
@@ -487,7 +488,36 @@ private CallbackResponse submitClaim(CallbackParams callbackParams) {
             populateWithPartyIds(dataBuilder);
         }
 
+        List<String> errors = new ArrayList<>();
+        if (caseData.getSdtRequestIdFromSdt() != null) {
+            List<String> postcodes = new ArrayList<>();
+            postcodes.add(caseData.getApplicant1().getPrimaryAddress().getPostCode());
+            postcodes.add(caseData.getRespondent1().getPrimaryAddress().getPostCode());
+            if (caseData.getRespondent2() != null) {
+                postcodes.add(caseData.getRespondent2().getPrimaryAddress().getPostCode());
+            }
+            postcodes.forEach(postcode -> {
+                if (!postcodeValidator.validate(postcode).isEmpty()) {
+                    errors.add("Postcode error, bulk claim");
+                }
+            });
+            // assign StdRequestId, to ensure duplicate requests from SDT/bulk claims are not processed
+            List<Element<String>> stdRequestIdList = new ArrayList<>();
+            stdRequestIdList.add(element(caseData.getSdtRequestIdFromSdt()));
+            dataBuilder.sdtRequestId(stdRequestIdList);
+            //TODO implement bulk claims that have interest added.
+            BigDecimal interest = interestCalculator.calculateInterest(caseData);
+            dataBuilder.claimFee(feesService.getFeeDataByTotalClaimAmount(caseData.getTotalClaimAmount().add(interest)));
+            //PBA manual selection
+            List<String> pbaNumbers = getPbaAccounts(callbackParams.getParams().get(BEARER_TOKEN).toString());
+            dataBuilder.applicantSolicitor1PbaAccounts(DynamicList.builder()
```

Comment:
We are doing the same thing in calculateFee method. But implementation is a bit different. Could you check please

Reply:
For bulk claims we don't have a actual user to select the PBA from the list of PBAs for the given organisation.

And for now at least we are not/were given any specific rationale on how to determine which one to set manually.

So we do as you said, assign the selected PBA to whatever PBA is first